ludwig/visualize.py

```diff
@@ -3055,7 +3074,24 @@ class and all the others will be considered negative. `positive_label` is
     if not isinstance(ground_truth, np.ndarray):
         # not np array, assume we need to translate raw value to encoded value
         feature_metadata = metadata[output_feature_name]
-        ground_truth = _vectorize_ground_truth(ground_truth, feature_metadata["str2idx"], ground_truth_apply_idx)
+        if "str2idx" in feature_metadata:
+            # categorical output feature as binary
+            ground_truth = _vectorize_ground_truth(ground_truth, feature_metadata["str2idx"], ground_truth_apply_idx)
+        else:
+            # binary output feature
+            if "str2bool" in feature_metadata:
+                # non-standard boolean representation
+                ground_truth = _vectorize_ground_truth(
+                    ground_truth, feature_metadata["str2bool"], ground_truth_apply_idx
+                )
+            else:
+                # standard boolean representation
+                ground_truth = ground_truth.values
+
+            # convert to 0/1 representation
+            ground_truth = ground_truth.astype(np.int)
+            # ensure positive_label is 1 for binary feature
+            positive_label = 1
```

Comment:
This code looks very similar / duplicated as the one from above. Could we factor this out into a separate method and call it in both places?

Reply:
got it...wilco.

Reply:
@justinxzhao factored out common code into a callable function.